module/zfs/dsl_crypt.c

```diff
@@ -332,8 +332,10 @@ spa_keystore_fini(spa_keystore_t *sk)
 static int
 dsl_dir_get_encryption_root_ddobj(dsl_dir_t *dd, uint64_t *rddobj)
 {
-	if (dd->dd_crypto_obj == 0)
+	if (dd->dd_crypto_obj == 0) {
+		dprintf("%s:%d: Returning %d\n", __func__, __LINE__, ENOENT);
```

Comment:
There is `zfs_flags` which has these bits:
https://github.com/openzfsonwindows/openzfs/blob/windows/include/sys/zfs_debug.h#L46-L59

In particular, setting `1` (should) turn on dprintf but at the moment I believe I have it permanently on.
and `512` (ZFS_DEBUG_SET_ERROR) which kicks the `SET_ERROR()` macro (on the next line), which will dprintf if the value is non-zero.
So there should be no reason to add the dprintfs for return code, at least we can avoid them in the shared files.

It would be set with; `kstat.exe -w zfs:0:tunable:zfs_flags=513` 




https://github.com/openzfsonwindows/openzfs/blob/windows/include/os/windows/spl/sys/sysmacros.h#L136

well that's disappointing - let's address that :)

Reply:
Thanks for pointing out

-> I will replace these dprintf's with TraceEvent and define TraceEvent as (void)0 in debug.h for both release and debug build. Will it work ?

Reply:
Raised new PR https://github.com/openzfsonwindows/openzfs/pull/38 that does not contain common file changes